core/string/ustring.h

```diff
@@ -68,12 +72,12 @@ class CharProxy {
 		return _cowdata.ptr() + _index;
 	}
 
-	_FORCE_INLINE_ void operator=(const T &other) const {
-		_cowdata.set(_index, other);
+	_FORCE_INLINE_ void operator=(const T &p_other) const {
+		_cowdata.set(_index, p_other);
 	}
 
-	_FORCE_INLINE_ void operator=(const CharProxy<T> &other) const {
-		_cowdata.set(_index, other.operator T());
+	_FORCE_INLINE_ void operator=(const CharProxy<T> &p_other) const {
+		_cowdata.set(_index, p_other.operator T());
```

Comment:
@hpvb @bruvzg With @jordo we wondered if there was an error in this copy assignment operator with the use of `_index`. Should it be `p_other._index` instead (equivalent to what the new copy assignment operator does)?

Reply:
Ya, if you are assigning this proxy, don't you want to grab the index from the other?    When you assign to a different `CowData` (function above this one), then you would want to retain your index.  But not assignment to another `CharProxy<T>`?

Reply:
Is it used at all? A write proxy of the Vector consists of a single [] operator, not sure why  the one for String is more complex.

Reply:
If I try to remove it, it gives an error here:
```
editor/plugins/canvas_item_editor_plugin.cpp: In member function 'void CanvasItemEditorViewport::_create_nodes(Node*, Node*, String&, const Point2&)':
editor/plugins/canvas_item_editor_plugin.cpp:5826:52: error: use of deleted function 'CharProxy<char32_t>& CharProxy<char32_t>::operator=(const CharProxy<char32_t>&)'
 5826 |                         name[0] = name.to_lower()[0];
      |                                                    ^
In file included from ./core/templates/hash_map.h:37,
                 from ./core/object/object.h:38,
                 from ./core/variant/binder_common.h:35,
                 from ./core/object/method_bind.h:34,
                 from ./core/object/class_db.h:34,
                 from ./core/object/undo_redo.h:34,
                 from ./editor/editor_data.h:34,
                 from ./editor/editor_node.h:35,
                 from editor/plugins/canvas_item_editor_plugin.h:34,
                 from editor/plugins/canvas_item_editor_plugin.cpp:31:
./core/string/ustring.h:45:7: note: 'CharProxy<char32_t>& CharProxy<char32_t>::operator=(const CharProxy<char32_t>&)' is implicitly deleted because the default definition would be ill-formed:
   45 | class CharProxy {
      |       ^~~~~~~~~
./core/string/ustring.h:45:7: error: non-static const member 'const int CharProxy<char32_t>::_index', cannot use default assignment operator
./core/string/ustring.h:45:7: error: non-static reference member 'CowData<char32_t>& CharProxy<char32_t>::_cowdata', cannot use default assignment operator
```

Reply:
It seems to be working correctly.

Something like `a[3] = b[5]`, will have `_index = 3` and `other._index = 5`, `_index` is used as write position, and `other._index` is handled in the `operator ()` of the `other` to get read position.

If index is set (which also require to change it to `mutable`, operator is `const`), results looks broken:
```c++
String caaname = "ABCDE";

//Index is NOT set (current behaviour):
caaname[3] = caaname.to_lower()[2]; // ==> ABCcE
//                                           ^^
//Index is set:
caaname[3] = caaname.to_lower()[2]; // ==> ABcDE
//                                           ^^
```